Packs/PaloAltoNetworks_Threat_Vault/Integrations/Threat_Vault/Threat_Vault.py

```diff
@@ -0,0 +1,485 @@
+
+    def ip_geo_get_request(self, ip_: str) -> dict:
+        """Get IP geolocation by sending a GET request.
+
+        Args:
+            ip_: ip address.
+        Returns:
+            Response from API.
+        """
+        return self._http_request(method='GET', url_suffix=f'/ip/{ip_}/geolocation', params=self._params)
+
+    def search_request(self, path: str, from_: int, to_: int, signature_name: str = '', domain_name: str = '',
+                       vendor: str = '', cve: str = '') -> dict:
+        """Initiate a search by sending a POST request.
+
+        Args:
+            path: API endpoint path to search.
+            from_: from which signature to return results
+            to_: to which signature to return results
+            signature_name: signature name.
+            domain_name: domain name.
+            vendor: vendor ID.
+            cve: cve ID.
+        Returns:
+            Response from API.
+        """
+        if path == 'dns':  # DNS search
+            if signature_name and domain_name:
+                raise Exception('Please provide either a signature_name or a domain_name.')
+        elif path == 'ips':  # Anti spyware search
+            if (cve and (vendor or signature_name)) or (vendor and (cve and signature_name)) \
+                    or (signature_name and (cve or vendor)):
+                raise Exception('Please provide either a signature_name or a cve or a vendor.')
+
+        data: Dict[str, Any] = {
+            'from': from_,
+            'size': to_ - from_
+        }
+        if signature_name:
+            data['field'] = 'signatureName'
+            data['value'] = signature_name
+        elif domain_name:
+            data['field'] = 'domainName'
+            data['value'] = domain_name
+        elif cve:
+            data['field'] = 'cve'
+            data['value'] = cve
+        else:  # vendor name
+            data['field'] = 'vendor'
+            data['value'] = vendor
+
+        return self._http_request(method='POST', url_suffix=f'/threatvault/{path}/search', params=self._params,
+                                  json_data=data)
+
+    def signature_search_results_request(self, search_type: str, search_request_id: str) -> dict:
+        """Get signature search results by sending a GET request.
+
+        Args:
+            search_type: search type.
+            search_request_id: signature id.
+        Returns:
+            Response from API.
+        """
+        return self._http_request(method='GET',
+                                  url_suffix=f'/threatvault/{search_type}/search/result/{search_request_id}',
+                                  params=self._params)
+
+
+def test_module(client: Client, *_) -> str:
+    """Performs basic get request to get a DNS signature.
+
+    Args:
+        client: Client object with request.
+
+    Returns:
+        string.
+    """
+    client.dns_signature_get_request(dns_signature_id='325235352')
```

Comment:
is this stable and will not get 404 in time?

Reply:
I believe so, the signature ID should not change, will change it to an IP anyhow.